tests/suites/test_suite_x509write.data

```diff
@@ -87,19 +87,85 @@ depends_on:MBEDTLS_SHA1_C:MBEDTLS_RSA_C:MBEDTLS_PKCS1_V15:MBEDTLS_DES_C:MBEDTLS_
+X509 String to Names #9 (Escape non-allowed characters)
 mbedtls_x509_string_to_names:"C=NL, O=Offspark\a Inc., OU=PolarSSL":"":MBEDTLS_ERR_X509_INVALID_NAME
 
-X509 String to Names #6 (Escape at end)
+X509 String to Names #10 (Escape at end)
 mbedtls_x509_string_to_names:"C=NL, O=Offspark\":"":MBEDTLS_ERR_X509_INVALID_NAME
+
+X509 String to Names #11 (unterminated quote)
+mbedtls_x509_string_to_names:"C=NL, O="Offspark":"":MBEDTLS_ERR_X509_INVALID_NAME
+
+X509 String to Names #12 (empty last name)
+mbedtls_x509_string_to_names:"C=NL, O=":"":MBEDTLS_ERR_X509_INVALID_NAME
+
+x509 String to Names #13 (empty name)
+mbedtls_x509_string_to_names:"C=, O=Offspark":"":MBEDTLS_ERR_X509_INVALID_NAME
+
+x509 String to Names #14 (empty quoted name)
+mbedtls_x509_string_to_names:"C="", O=Offspark":"":MBEDTLS_ERR_X509_INVALID_NAME
+
+x509 String to Names #15 (optional white spaces after end quote)
+mbedtls_x509_string_to_names:"C=NL,O="Offspark, Inc."    , OU=PolarSSL":"C=NL, O=Offspark\, Inc., OU=PolarSSL":0
+
+x509 String to Names #16 (optional white spaces before start quote)
+mbedtls_x509_string_to_names:"C=NL,O=    "Offspark, Inc.", OU=PolarSSL":"C=NL, O=Offspark\, Inc., OU=PolarSSL":0
+
+X509 String to Names #17 (optional white spaces after equal)
+mbedtls_x509_string_to_names:"C=NL,O=     Offspark\, Inc., OU=PolarSSL":"C=NL, O=Offspark\, Inc., OU=PolarSSL":0
+
+X509 String to Names #18 (optional white spaces before equal)
+mbedtls_x509_string_to_names:"C=NL,O     =Offspark\, Inc., OU=PolarSSL":"C=NL, O=Offspark\, Inc., OU=PolarSSL":0
+
+X509 String to Names #19 (preceding white space before first tag)
+mbedtls_x509_string_to_names:" C=NL,O=Offspark\, Inc., OU=PolarSSL":"C=NL, O=Offspark\, Inc., OU=PolarSSL":0
+
+X509 String to Names #20 (invalid whitespace within tag)
+mbedtls_x509_string_to_names:"C=NL,O=Offspark\, Inc., O U=PolarSSL":"":MBEDTLS_ERR_X509_INVALID_NAME
+
+X509 String to Names #21 (input with all spaces)
+mbedtls_x509_string_to_names:"   ":"":MBEDTLS_ERR_X509_INVALID_NAME
+
+X509 String to Names #22 (value with all spaces)
+mbedtls_x509_string_to_names:"C=   ":"":MBEDTLS_ERR_X509_INVALID_NAME
+
+X509 String to Names #23 (valid tag with only spaces)
+mbedtls_x509_string_to_names:"C   ":"":MBEDTLS_ERR_X509_INVALID_NAME
+
+X509 String to Names #24 (only comma separated valid tags)
+mbedtls_x509_string_to_names:"C,O,OU":"":MBEDTLS_ERR_X509_INVALID_NAME
+
+X509 String to Names #25 (comma after valid tag)
+mbedtls_x509_string_to_names:"C,O=Arm":"":MBEDTLS_ERR_X509_INVALID_NAME
+
+X509 String to Names #26 (invalid tag)
+mbedtls_x509_string_to_names:"C=NL,ABC=BAD, OU=PolarSSL":"":MBEDTLS_ERR_X509_UNKNOWN_OID
+
+X509 String to Names #27 (name with long trailing whitespaces)
+mbedtls_x509_string_to_names:"C=NL                                                                                                                                                                                                                                    ":"C=NL":0
```

Comment:
@RonEld: You pointed out that this test covers the case case of very long strings. However, I think the data buffer is of length MBEDTLS_X509_MAX_DN_NAME_SIZE which is hard-coded to 256 by default. But this test only has a DN of length 232 so I do not think the buffer will actually be overflowed. Perhaps I missed something?

I think it would also be good to have a test has a value of length exactly 256 and the character at position 255 is a non whitespace. This is just a corner case and it would be good to make sure we do not miss any data

Reply:
my mistake in counting whitespaces